src/hooks/use-view-service.ts

```diff
@@ -1,8 +1,13 @@
 import { useMemo } from "@nebula.js/stardust";
 import createViewService from "../services/view-service";
+import type { PivotLayout } from "../types/QIX";
 import type { PageInfo, ViewService } from "../types/types";
 
-// eslint-disable-next-line react-hooks/exhaustive-deps
-const useViewService = (pageInfo: PageInfo): ViewService => useMemo(() => createViewService(), [pageInfo.page]);
+const useViewService = (layout: PivotLayout, pageInfo: PageInfo): ViewService =>
+  useMemo(
+    () => createViewService(layout.snapshotData),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [layout.snapshotData, pageInfo.page],
```

Comment:
I'm not sure pageInfo.page should be in the dependencies. It was there before, so I kept it as it is.

Reply:
`pageInfo.page` in deps means that the view service is "reset" or re-created when page changes. Which is what we want since all states in viewService are based on the current page.